### ISSUE TITLE
Warn about :: _ partial type signatures

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/hlint#331]
People sometimes put them in for debugging - should have a lint check for them.

**Comment by mitchellwrosen** [comment]
This is already a GHC warning `-Wpartial-type-signatures`, should it also be detected by hlint?

**Comment by ndmitchell** [comment]
GHC warns about partial type signatures if a type signature has `_` anywhere in it, so `x :: _ -> _` triggers that warning. I'm thinking about a lint rule exclusively for `x :: _`, where the `_` is completely literal. There are some good reasons for partial type signatures, but never a good reason for `:: _`, other than so it shows up in your IDE, for debugging - in which case people don't usually intend to commit it.

**Comment by ndmitchell** [comment]
What about `@_` type applications? Are they even legal?

**Comment by mitchellwrosen** [comment]
They are legal! You'd use them to let inference figure out some type argument, like so:

```haskell
foo @_ @_ @Int
```

**Comment by ndmitchell** [comment]
Ah, so useful, unless they are the final argument in an application sequence.
